### PR TITLE
Split pipeline overview and config file examples out of Getting Started

### PR DESCRIPTION
**Pull request by debadair**
I changed the heading of the event lifecycle topic to Logstash Processing Pipeline, and changed the config file format topic to Configuring Logstash. I merged the topics removed from Getting Started into those two files.

I think the order of the topics should be tweaked so it's:

getting started with logstash
logstash processing pipeline
configuring logstash
command-line flags
extending logstash
adding a sample filter to logstash
input plugins
output plugins
filter plugins
codec plugins

Not sure how to do that?

### CODE REVIEW

**Comment by suyograo** [comment]
@debadair for the index, we can change the order directly here: https://github.com/elasticsearch/logstash-docs/blob/master/docs/index.asciidoc

So, to give you some background, the way doc generation works is we keep this repo as source of truth for all static docs. Each plugin in LS will generate its own asciidoc and there is an orchestration script which merges the plugin docs and static docs. These combined docs are then placed in https://github.com/elasticsearch/logstash-docs which is an intermediate holding place. Doc generation script then converts them to html and uploads to es.org website :)

File: docs/asciidoc/static/configuration.asciidoc [hunk 1]
```diff
@@ -417,3 +416,290 @@ output {
...
+
+
+[[config-examples]]
+=== Logstash Configuration Examples
+The following examples illustrate how you can configure Logstash to filter events, process Apache logs and syslog messages, and use conditionals to control what events are processed by a filter or output.
+
+[float]
+[[filter-example]]
+==== Configuring Filters
+Filters are an in-line processing mechanism that provide the flexibility to slice and dice your data to fit your needs. Let's take a look at some filters in action. The following configuration file sets up the *grok* and *date* filters.
```
Comment (suyograo) [inline docs/asciidoc/static/configuration.asciidoc]: I think grok and date should be wrapped in backticks

File: docs/asciidoc/static/configuration.asciidoc [hunk 2]
```diff
@@ -417,3 +416,290 @@ output {
...
+
+The other filter used in this example is the *date* filter. This filter parses out a timestamp and uses it as the timestamp for the event (regardless of when you're ingesting the log data). You'll notice that the @timestamp field in this example is set to December 11, 2013, even though Logstash is ingesting the event at some point afterwards. This is handy when backfilling logs. It gives you the ability to tell Logstash "use this value as the timestamp for this event".
```
Comment (suyograo) [inline docs/asciidoc/static/configuration.asciidoc]: @timestamp should be wrapped in backticks

**Comment by suyograo** [comment]
minor comments about formatting, but LGTM otherwise

**Comment by kurtado** [comment]
It would be great to get some separation and sub-headers between the links. Kindofa bit of a mish-mash currently. (My bad on not including that from the start :) 

Maybe group into sections, à la: 
* Running/Using Logstash
* Plugin docs
* Logstash for Developers

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**